sdcm/nemesis.py

```diff
@@ -3771,8 +3774,11 @@ def remove_node():
             assert removed_node_status is None, \
                 "Node was not removed properly (Node status:{})".format(removed_node_status)
 
-            # add new node
-            new_node = self._add_and_init_new_cluster_nodes(count=1, rack=self.target_node.rack)[0]
+            # add new node with same type (data node / zero token node)
+            new_node_args = {"count": 1, "rack": self.target_node.rack}
+            if self.target_node._is_zero_token_node:
```

Comment:
not sure how self.target_node is still available after remove/decommission? 

did you test this change

Reply:
self.target_node will be available as python object after node as instance will be deleted. the property node._is_zero_token_node is set on node initialization, and not counted. So this property will be always available.

Yes this changes were tested. i will updated with results